### PR TITLE
WebIDL Constructors: add tests to constructors with required arguments which, when called as functions, throw an error

### PR DESCRIPTION
**Pull request by JakeChampion**
This cannot be tested generically (see #34157 for work on that) as the first argument is required.

### CODE REVIEW

**Review by domenic — APPROVED** [review]

**Comment by JakeChampion** [comment]
@domenic and @annevk , I am doing this for all the constructors which have required arguments - should I do these as separate pull-requests?

**Comment by domenic** [comment]
I think a single PR would be easier to review, personally!

However, I'm not sure if we need that extra coverage for every required-arg constructor. We can probably assume that if a browser implements no-required-arg constructors correctly in this regard, they'll also get required-arg constructors correct.

In cases like this, where there are already tests for the constructor, just adding one more line to the file is reasonable. But I'm not sure we need to do it everywhere.

**Comment by JakeChampion** [comment]
> I think a single PR would be easier to review, personally!
> 
> However, I'm not sure if we need that extra coverage for every required-arg constructor. We can probably assume that if a browser implements no-required-arg constructors correctly in this regard, they'll also get required-arg constructors correct.
> 
> In cases like this, where there are already tests for the constructor, just adding one more line to the file is reasonable. But I'm not sure we need to do it everywhere.

Would it be okay if I push the commits I already have done for a bunch of the required arg constructors to this branch/pull-request?

**Comment by domenic** [comment]
Sure, no problem.

File: eventsource/dedicated-worker/eventsource-constructor-no-new.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+test(function() {
```
Comment (foolip) [inline eventsource/dedicated-worker/eventsource-constructor-no-new.js:1]: This file won't be recognized a test, you'll need to name it .any.js or .window.js

Comment (JakeChampion) [reply eventsource/dedicated-worker/eventsource-constructor-no-new.js]: Thanks @foolip

**Review by hoch — APPROVED** [review]
webaudio/resources/audit.js LGTM

**Comment by JakeChampion** [comment]
@domenic I've rebased this and believe it is ready for another round of reviews 👍

File: html/canvas/element/drawing-text-to-the-canvas/2d.text.draw.fill.maxWidth.large.png
Comment (domenic) [inline html/canvas/element/drawing-text-to-the-canvas/2d.text.draw.fill.maxWidth.large.png:1]: I don't know why this file was changed.

File: css/css-ui/webkit-appearance-auto-non-html-namespace-001.html
Comment (domenic) [inline css/css-ui/webkit-appearance-auto-non-html-namespace-001.html:1]: This file seems likely added by mistake?

Comment (JakeChampion) [reply css/css-ui/webkit-appearance-auto-non-html-namespace-001.html:1]: It appears if I run `./css/css-ui/tools/appearance-build-webkit-reftests.py`
I've removed the file and rebased 👍

File: html/canvas/element/drawing-text-to-the-canvas/2d.text.draw.fill.maxWidth.large.png
Comment (JakeChampion) [reply html/canvas/element/drawing-text-to-the-canvas/2d.text.draw.fill.maxWidth.large.png:1]: Me neither - I've restored the original 👍

File: webstorage/event_constructor.window.js [hunk 2]
```diff
@@ -1,3 +1,11 @@
+test(function() {
+    assert_throws_js(
+        TypeError,
+        () => StorageEvent(""),
+        "Calling StorageEvent constructor without 'new' must throw"
+    );
```
Comment (inexorabletash) [inline webstorage/event_constructor.window.js:6]: Add test description

File: html/webappapis/scripting/events/messageevent-constructor.https.html [hunk 3]
```diff
@@ -4,6 +4,14 @@
 <script src="/resources/testharnessreport.js"></script>
 <script src="/service-workers/service-worker/resources/test-helpers.sub.js"></script>
 <script>
+test(function() {
+    assert_throws_js(
+        TypeError,
+        () => MessageEvent(""),
+        "Calling MessageEvent constructor without 'new' must throw"
+    );
+});
```
Comment (inexorabletash) [inline html/webappapis/scripting/events/messageevent-constructor.https.html:13]: Add test description

File: html/webappapis/scripting/processing-model-2/unhandled-promise-rejections/promise-rejection-event-constructor.html [hunk 4]
```diff
@@ -9,6 +9,12 @@
 test(function() {
   var p = new Promise(function(resolve, reject) {});
 
+  assert_throws_js(TypeError,
+  function() {
```
Comment (inexorabletash) [inline html/webappapis/scripting/processing-model-2/unhandled-promise-rejections/promise-rejection-event-constructor.html:13]: Indent the arguments to `assert_throws_js` as is done later in the file.

File: webmessaging/broadcastchannel/basics.any.js [hunk 5]
```diff
@@ -1,3 +1,11 @@
+test(function() {
+  assert_throws_js(
+      TypeError,
+      () => BroadcastChannel(""),
+      "Calling BroadcastChannel constructor without 'new' must throw"
+    );
+});
```
Comment (inexorabletash) [inline webmessaging/broadcastchannel/basics.any.js:7]: Add description

**Review by inexorabletash — COMMENTED** [review]
Overall LGTM, just lots of nitpicks. Sorry for being a broken record about the test name, but it's important when reviewing the results when something does go wrong.

**Comment by JakeChampion** [comment]
@inexorabletash Thanks for the thorough review - I believe I have addressed all the feedback now ☺️

**Review by inexorabletash — APPROVED** [review]
LGTM, thank you!!!

**Pull request merged**